server/src/test/java/io/crate/integrationtests/AlterTableIntegrationTest.java

```diff
@@ -25,31 +25,38 @@
 import static io.crate.testing.Asserts.assertThrowsMatches;
 import static io.crate.testing.SQLErrorMatcher.isSQLError;
 import static io.netty.handler.codec.http.HttpResponseStatus.BAD_REQUEST;
-import static org.hamcrest.Matchers.is;
+import static org.hamcrest.Matchers.containsString;
 import static org.hamcrest.Matchers.startsWith;
 
 import org.junit.Test;
 
-
 public class AlterTableIntegrationTest extends SQLIntegrationTestCase {
 
     @Test
-    public void test_alter_soft_delete_setting_for_partitioned_tables() {
-        execute("create table test(i int) partitioned by (i)");
-        assertThat(response.rowCount(), is(1L));
-        execute("alter table test set (\"soft_deletes.enabled\" = false)");
-
+    public void test_create_soft_delete_setting_for_partitioned_tables() {
         assertThrowsMatches(
-            () -> execute("insert into test values(1)"),
+            () -> execute("create table test(i int) partitioned by (i) WITH(\"soft_deletes.enabled\" = false) "),
             isSQLError(startsWith("Creating tables with soft-deletes disabled is no longer supported."),
                        INTERNAL_ERROR,
                        BAD_REQUEST,
                        4000)
         );
+    }
 
-        execute("alter table test set (\"soft_deletes.enabled\" = true)");
-        execute("insert into test values(1)");
-        assertThat(response.rowCount(), is(1L));
+    @Test
+    public void test_alter_table_update_final_setting_on_open_table() {
```

Comment:
not sure how to replace these tests by unit tests - this check happens in executor.

it's easier with dynamic/non-dynamic cases as they used to fail during analysis phase
but analyse(alter some_final_setting) will pass here
and it would not be a representative test as it actually throws an exception later.

Reply:
Final settings should be filtered out at the `TableParameters`. This way exception will be raised while analyzing and tests can be moved into unit tests.